--- a/unisocwcn/Makefile
+++ b/unisocwcn/Makefile
@@ -79,9 +79,9 @@ ifeq ($(CONFIG_AW_WIFI_DEVICE_UWE5622),y)
 #### add cflag ######
 export CONFIG_WCN_SDIO = y
 #export CONFIG_WCN_USB = y
 # export CONFIG_WCN_GNSS = y
-ccflags-y += -DCONFIG_CHECK_DRIVER_BY_CHIPID
-# ccflags-y += -DCONFIG_UWE5622
+#ccflags-y += -DCONFIG_CHECK_DRIVER_BY_CHIPID
+ccflags-y += -DCONFIG_UWE5622
 BSP_CHIP_ID := uwe5622
 WCN_HW_TYPE := sdio
 endif
