--- a/unisocwcn/platform/wcn_parn_parser.c
+++ b/unisocwcn/platform/wcn_parn_parser.c
@@ -178,7 +178,7 @@ static int parse_firmware_path(char *firmware_path)
 			continue;
 		}
 		memset(fstab_name, 0, sizeof(fstab_name));
-		strncpy(fstab_name, fstab_dir[loop], sizeof(fstab_dir[loop]));
+		strncpy(fstab_name, fstab_dir[loop], sizeof(fstab_name) - 1);
 		if (strlen(fstab_name) > 1)
 			fstab_name[strlen(fstab_name)] = '/';
 		iterate_dir(file1, &ctx);
