--- a/unisocwcn/platform/rdc_debug.c
+++ b/unisocwcn/platform/rdc_debug.c
@@ -42,17 +42,17 @@ static unsigned int wcn_cp2_log_limit_size =
 static unsigned int wcn_cp2_log_cover_old = 1;
 /* path of config file unisoc_cp2log_config.txt */
-#define WCN_DEBUG_CFG_MAX_PATH_NUM	0
+#define WCN_DEBUG_CFG_MAX_PATH_NUM	2
 static char *wcn_cp2_config_path[WCN_DEBUG_CFG_MAX_PATH_NUM] = {
 	"/data/unisoc_cp2log_config.txt",
 	"/vendor/etc/wifi/unisoc_cp2log_config.txt"
 };
 /* path of cp2 log and mem files. */
-#define WCN_UNISOC_DBG_MAX_PATH_NUM	0
+#define WCN_UNISOC_DBG_MAX_PATH_NUM	3
 static char *wcn_unisoc_dbg_path[WCN_UNISOC_DBG_MAX_PATH_NUM] = {
 	UNISOC_DBG_PATH_DEFAULT,/* most of projects */
 	"/data",		/* amlogic s905w... */
 	"/mnt/UDISK"		/* allwinner r328... */
 };
 
 #define WCN_CP2_LOG_NAME	"/unisoc_cp2log_%%d.txt"
