--- a/unisocwifi/Makefile
+++ b/unisocwifi/Makefile
@@ -22,6 +22,14 @@ MODULE_NAME := sprdwl_ng
 endif
 
 ####add cflag######
+ccflags-y += -Wno-implicit-fallthrough
+ccflags-y += -Wno-attribute-warning
+ccflags-y += -Wno-discarded-qualifiers
+ccflags-y += -Wno-incompatible-pointer-types
+ccflags-y += -Wno-vla
+ccflags-y += -Wno-unused-variable
+ccflags-y += -Wno-enum-conversion
+ccflags-y += -D_FORTIFY_SOURCE=0
 ccflags-y += -DUWE5621_FTR
 ccflags-y += -DIBSS_SUPPORT -DIBSS_RSN_SUPPORT
 ccflags-y += -DNAN_SUPPORT
