--- a/unisocwifi/cfg80211.c
+++ b/unisocwifi/cfg80211.c
@@ -1527,8 +1527,7 @@ static int sprdwl_cfg80211_scan(struct 
 			if (!ssids[i].ssid_len)
 				continue;
 			scan_ssids->len = ssids[i].ssid_len;
-			strncpy(scan_ssids->ssid, ssids[i].ssid,
-				ssids[i].ssid_len);
+			memcpy(scan_ssids->ssid, ssids[i].ssid, ssids[i].ssid_len);
 			scan_ssids_len += (ssids[i].ssid_len
 					   + sizeof(scan_ssids->len));
 			scan_ssids = (struct sprdwl_scan_ssid *)
