--- a/unisocwifi/Makefile
+++ b/unisocwifi/Makefile
@@ -32,7 +32,7 @@ ccflags-y += -DCOMPAT_SAMPILE_CODE
 ccflags-y += -DRND_MAC_SUPPORT
 #ccflags-y += -DSOFTAP_HOOK
 ccflags-y += -DATCMD_ASSERT
-ccflags-y += -DTCPACK_DELAY_SUPPORT
-#ccflags-y += -DDFS_MASTER
+ccflags-y += -DTCPACK_DELAY_SUPPORT
+ccflags-y += -DDFS_MASTER
 #ccflags-y += -DRX_NAPI
 ifneq ($(TARGET_BUILD_VARIANT),user)
@@ -67,7 +67,7 @@ $(MODULE_NAME)-objs += rf_marlin3.o
 $(MODULE_NAME)-objs += rtt.o
 $(MODULE_NAME)-objs += api_version.o
 $(MODULE_NAME)-objs += rnd_mac_addr.o
 $(MODULE_NAME)-objs += debug.o
-#$(MODULE_NAME)-objs += 11h.o
+$(MODULE_NAME)-objs += 11h.o
 
 all: $(all_dependencies)
 driver: $(driver_dependencies)
